Handbook: Update docs ownership + minor tweak to gardener notes (#11575)
PostHog/posthog.com: contents/handbook/content-and-docs/docs.md

The docs gardener is a critical role for ensuring our docs are up to date and useful. We assign someone from the Content & Docs team to do this each week. Their responsibility is to:

1. Review the feedback across the channels below.
2. Create issues for docs updates with the context on why it's important and the source of the feedback (unless it is a very simple fix). Be careful not to publish private data (like user emails) in these. 
3. Assign the issue to the [docs owner](#ownership-in-content--docs-team).

> 4. When done, attach a ✅ emoji or similar (if possible) so that the next docs gardener knows that this has been assigned.

Reviewer: Replying on Lior's behalf, but I think your assumption is correct. Basically: tick it Slack so we don't double up.

> 4. When done, attach a ✅ emoji or similar (if possible) so that the next docs gardener knows that this has been assigned.

Reviewer 2: ```suggestion
4. When done, add a ✅ emoji react to the message in Slack or similar (if possible) so that the next docs gardener knows that this has been assigned.
```

1. Review the feedback across the channels below.

> 2. Create issues for docs updates with the context on why it's important and the source of the feedback (unless it is a very simple fix).

Reviewer 2: ```suggestion
2. Create issues for docs updates with the context on why it's important and the source of the feedback (unless it is a very simple fix). Be careful not to publish private data (like user emails) in these. 
```

3. Assign the issue to the [docs owner](#ownership-in-content--docs-team).

> 4. When done, attach a ✅ emoji or similar (if possible) so that the next doc's gardener knows that this has been assigned.

Reviewer 2: ```suggestion
4. When done, attach a ✅ emoji or similar (if possible) so that the next docs gardener knows that this has been assigned.
```
